src/node_worker.cc

```diff
@@ -58,6 +58,44 @@ void WaitForWorkerInspectorToStop(Environment* child) {
 
 }  // anonymous namespace
 
+void LoopStopper::Install(Environment* env, void* data, uv_async_cb target) {
+  env_ = env;
+  async_ = new uv_async_t;
+  if (data != nullptr) async_->data = data;
+  CHECK_EQ(uv_async_init(env_->event_loop(), async_, target), 0);
+}
+
+void LoopStopper::Uninstall() {
+  Mutex::ScopedLock lock(mutex_);
+  if (async_ != nullptr) {
+    env_->CloseHandle(async_, [](uv_async_t* async) { delete async; });
+  }
+}
+
+void LoopStopper::Stop() {
+  Mutex::ScopedLock lock(mutex_);
+  stop_ = true;
+  if (async_ != nullptr) uv_async_send(async_);
+}
+
+void LoopStopper::SetStopped() {
+  Mutex::ScopedLock lock(mutex_);
+  stop_ = true;
+}
+
+bool LoopStopper::IsStopped() {
+  Mutex::ScopedLock lock(mutex_);
+  return stop_;
+}
+
+uv_async_t* LoopStopper::GetHandle() {
+  Mutex::ScopedLock lock(mutex_);
+  return async_;
+}
+void LoopStopper::MemoryInfo(MemoryTracker* tracker) const {
+  tracker->TrackField("loop_stopper_async", *async_);
```

Comment:
I think this might work because `TrackField()` only uses `sizeof` for the `async_`, but it might be better to check whether `async_` is a `nullptr` before reporting it? (And maybe lock the mutex here as well.)

Reply:
done, thanks.